examples/allocations.rs

```diff
@@ -0,0 +1,172 @@
+
+use clap::{ArgEnum, Parser};
+
+#[derive(Parser, Debug)]
+#[clap(author, version, about, long_about = None)]
+struct Args {
+    #[clap(arg_enum, value_parser, default_value = "all")]
+    mode: Mode,
+
+    #[clap(short, long, value_parser, default_value = "127.0.0.1:9042")]
+    node: String,
+
+    #[clap(short, long, value_parser, default_value_t = 256usize)]
+    parallelism: usize,
+
+    #[clap(short, long, value_parser, default_value_t = 100_000usize)]
+    requests: usize,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, ArgEnum)]
+enum Mode {
+    All,
+    Insert,
+    Select,
+}
+
+#[global_allocator]
+static GLOBAL: &StatsAlloc<System> = &INSTRUMENTED_SYSTEM;
+
+fn print_stats(stats: &stats_alloc::Stats, reqs: f64) {
+    println!(
+        "allocs/req:            {:9.2}",
+        stats.allocations as f64 / reqs
+    );
+    println!(
+        "reallocs/req:          {:9.2}",
+        stats.reallocations as f64 / reqs
+    );
+    println!(
+        "frees/req:             {:9.2}",
+        stats.deallocations as f64 / reqs
+    );
+    println!(
+        "bytes allocated/req:   {:9.2}",
+        stats.bytes_allocated as f64 / reqs
+    );
+    println!(
+        "bytes reallocated/req: {:9.2}",
+        stats.bytes_reallocated as f64 / reqs
+    );
+    println!(
+        "bytes freed/req:       {:9.2}",
+        stats.bytes_deallocated as f64 / reqs
+    );
+}
+
+async fn measure(
+    session: Arc<Session>,
+    prepared: Arc<PreparedStatement>,
+    sem: Arc<Semaphore>,
+    reqs: usize,
+    parallelism: usize,
+) -> Stats {
+    let initial_stats = GLOBAL.stats();
+
+    for i in 0..reqs {
+        if i % 10000 == 0 {
+            print!(".");
+            std::io::stdout().flush().unwrap();
+        }
+        let session = session.clone();
+        let prepared = prepared.clone();
+        let permit = sem.clone().acquire_owned().await;
+        tokio::task::spawn(async move {
+            let i = i;
+            session
+                .execute(&prepared, (i as i32, 2 * i as i32))
+                .await
+                .unwrap();
+
+            let _permit = permit;
+        });
+    }
+    println!();
+
+    // Wait for all in-flight requests to finish
+    for _ in 0..parallelism {
+        sem.acquire().await.unwrap().forget();
+    }
+
+    GLOBAL.stats() - initial_stats
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let args = Args::parse();
+    println!("{:?}", args);
+
+    println!("Connecting to {} ...", args.node);
+
+    let session: Session = SessionBuilder::new().known_node(args.node).build().await?;
+    let session = Arc::new(session);
+
+    session.query("CREATE KEYSPACE IF NOT EXISTS ks WITH REPLICATION = {'class' : 'SimpleStrategy', 'replication_factor' : 1}", &[]).await?;
+
+    session
+        .query(
+            "CREATE TABLE IF NOT EXISTS ks.alloc_test (a int, b int, c text, primary key (a, b))",
+            &[],
+        )
+        .await?;
+
```

Comment:
To be safe, you should wait for schema agreement after `CREATE KEYSPACE` and `CREATE TABLE`.

Reply:
You added `await_schema_agreement` after `CREATE TABLE`, but shouldn't it also be done after `CREATE KEYSPACE`? Correct me if I'm wrong, but if the driver sends `CREATE TABLE` statement to a different node than `CREATE KEYSPACE`, that node may not yet know about the new keyspace.

Reply:
Right. For this test it's rather a theoretical consideration, because it's expected to be run on a single Scylla instance, but I'll add another wait for completeness.

Reply:
done